sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -239,6 +288,36 @@ def __getitem__(self, key):
       # be more surprising than a clear error.
       raise frame_base.WontImplementError('non-deferred')
 
+  def keys(self):
+    return self.index
+
+  @frame_base.args_to_kwargs(pd.Series)
+  @frame_base.populate_defaults(pd.Series)
+  def append(self, to_append, ignore_index, verify_integrity, **kwargs):
+    if not isinstance(to_append, DeferredSeries):
+      raise frame_base.WontImplementError(
+          "append() only accepts DeferredSeries instances, received " +
+          str(type(to_append)))
+    if ignore_index:
+      raise frame_base.WontImplementError(
+          "append(ignore_index=True) is order sensitive")
+
+    if verify_integrity:
+      # verifying output has a unique index requires global index.
+      requires = partitionings.Singleton()
```

Comment:
Just a thought: it might be nice to attach a message as to why we chose Singleton here (that we could use when throwing an error). Fine with a TODO/JIRA.

Reply:
Yeah I was thinking about something like that as well. I'll file jira

Reply:
BEAM-11839